scrapi/base/helpers.py

```diff
@@ -114,28 +119,52 @@ def oai_process_uris(*args):
         elif arg:
             identifiers.append(arg)
 
+    return identifiers
+
+
+def gather_object_uris(identifiers):
     object_uris = []
-    provider_uris = []
     for item in identifiers:
         if 'doi' in item.lower():
-            doi = item.replace('doi:', '').replace('DOI:', '').strip()
-            if 'http://dx.doi.org/' in doi:
-                object_uris.append(doi)
-            else:
-                object_uris.append('http://dx.doi.org/{}'.format(doi))
+            url_doi, just_doi = URL_REGEX.search(item), DOI_REGEX.search(item)
+            url_doi = url_doi.group() if url_doi else None
```

Comment:
I think the group function I defined in my previous comment would be a good addition, since this logic is duplicated a few times, and it is a function that could be useful for other harvesters using regex matching.